sentry-android-core/src/test/java/io/sentry/android/core/ActivityFramesTrackerTest.kt

```diff
@@ -102,15 +110,166 @@ class ActivityFramesTrackerTest {
         assertEquals(slowFrames!!.value, 5f)
     }
 
+    @Test
+    fun `sets slow and frozen frames even if start was null`() {
+        val sut = fixture.getSut()
+        val arrayAll = SparseIntArray()
+        arrayAll.put(1, 100)
+        arrayAll.put(20, 5)
+        arrayAll.put(705, 6)
+        val array = arrayOf(arrayAll)
+
+        whenever(fixture.aggregator.metrics).thenReturn(null, array)
+
+        sut.addActivity(fixture.activity)
+        sut.setMetrics(fixture.activity, fixture.sentryId)
+
+        val metrics = sut.takeMetrics(fixture.sentryId)
+
+        val totalFrames = metrics!!["frames_total"]
+        assertEquals(totalFrames!!.value, 111f)
+
+        val frozenFrames = metrics["frames_frozen"]
+        assertEquals(frozenFrames!!.value, 6f)
+
+        val slowFrames = metrics["frames_slow"]
+        assertEquals(slowFrames!!.value, 5f)
+    }
+
+    @Test
+    fun `different activities have separate counts - even when called out of order`() {
+        val sut = fixture.getSut()
+        val array0 = SparseIntArray().also {
+            it.put(1, 100)
+            it.put(20, 5)
+            it.put(705, 6)
+        }.let { arrayOf(it) }
```

Comment:
`l`: A bit shorter
```suggestion
        val array0 = SparseIntArray().apply {
            put(1, 100)
            put(20, 5)
            put(705, 6)
        }.let { arrayOf(it) }
```

Reply:
Yeah I prefer `also` over `apply`. Fewer surprises this way. Gonna keep it this way if you don't mind :D

Reply:
No strong opinion. It's `l`.